src/masternodes/govvariables/attributes.cpp

```diff
@@ -134,39 +150,53 @@ UniValue ATTRIBUTES::Export() const {
 
 Res ATTRIBUTES::Validate(const CCustomCSView & view) const
 {
-    if (view.GetLastHeight() < Params().GetConsensus().FortCanningHillHeight) {
+    if (view.GetLastHeight() < Params().GetConsensus().FortCanningHillHeight)
         return Res::Err("Cannot be set before FortCanningHill");
-    }
 
     for (const auto& item : attributes) {
-        const auto strVec= KeyBreaker(item.first);
+        if (item.first.size() > 128) {
+            return Res::Err("Identifier exceed maximum length (128)");
+        }
+
+        const auto strVec = KeyBreaker(item.first);
 
         if (strVec.empty()) {
             return Res::Err("Empty map key found");
         }
 
-        // switch for the many more types coming later!
-        if (strVec[0].size() == 1 && strVec[0].at(0) == AttributeTypes::Token) {
-            if (strVec.size() != 3) {
-                return Res::Err("Three items expected in token key");
-            }
+        if (strVec[0].size() != 1) {
```

Comment:
Does not have to be 1, for token it is set to 1 but for others a longer string could be used.